src/Analyzer/PatternString.php

```diff
@@ -23,7 +23,7 @@ public function matches(string $pattern): bool
             return true;
         }
 
-        return fnmatch($pattern, $this->value, \FNM_NOESCAPE);
+        return 1 === preg_match('#^'.$this->convertShellToRegExPattern($pattern).'#', $this->value);
```

Comment:
This line can be difficult to read, what do you think about moving this condition into a specific function with a specific name to understand well what it does?

Reply:
Sure, done.
Now we could also move the following code inside the new method `startsWithPattern`:
```
if (!$this->containsWildcard($pattern) && str_starts_with($this->value, $pattern)) {
    return true;
}
```

Alternatively, unless for performance reasons, we could also remove the above code because it's already covered by this:
```
1 === preg_match('#^'.$this->convertShellToRegExPattern($pattern).'#', $this->value);
```

What do you think about this ?

Reply:
So, you prefer leave this code explicit and do not remove it, right?
```
if (!$this->containsWildcard($pattern) && str_starts_with($this->value, $pattern)) {
    return true;
}
```

```
    private function containsWildcard(string $pattern): bool
    {
        return
            str_contains($pattern, '*') ||
            str_contains($pattern, '?') ||
            str_contains($pattern, '.') ||
            str_contains($pattern, '[');
    }
```